testsuite/classlibrary/TestErrorHandling.sc

```diff
@@ -0,0 +1,41 @@
+
+
+TestThrow : UnitTest {
+
+	test_throw_nested_threads {
+		var condition = Condition.new;
+		var call1 = false, call2 = false;
+
+		fork { // isolate from surrounding thread
+			var r = Routine {
+				var r = Routine {
+					var r = Routine {
+						Error("test").throw;
+					};
+					var next = thisThread;
+					r.exceptionHandler = {
+						condition.unhang;
+						call1 = true;
+						next.handleError
+					};
+					r.next;
+					condition.hang;
+				};
+				var next = thisThread;
```

Comment:
I believe you can fix it by declaring an inner `condition = Condition.new` here.